apmproxy/receiver.go

```diff
@@ -87,9 +93,39 @@ func (c *Client) handleInfoRequest() (func(w http.ResponseWriter, r *http.Reques
 	customTransport.ResponseHeaderTimeout = c.client.Timeout
 	reverseProxy.Transport = customTransport
 
+	reverseProxy.ModifyResponse = func(rsp *http.Response) error {
+		if rsp.StatusCode != http.StatusOK {
+			b, err := io.ReadAll(rsp.Body)
+			if err != nil {
+				c.logger.Warnf("failed to read version response body: %v", err)
+				b = []byte{}
+			}
+
+			c.logger.Debugf("failed to query version from the APM server: response body: %s", string(b))
+
+			// Return an error to the ErrorHandler that will proxy the status code back to the agent.
+			return fmt.Errorf("failed to query version from the APM server: response status: %w", unexpectedStatusError(rsp.StatusCode))
```

Comment:
Why is this necessary? If we return nil here, then won't rsp.StatusCode already be set to whatever APM Server returned?

Reply:
Good point! This was added to increase visibility in the lambda extension, mostly for agents that do not log error reponse code/body.

Looking back at this, it's probably not appropriate for the lambda extension to touch the response. It should proxy it as is and leave that responsibility to the agent.

I've update the code to remove the the `ModifyResponse` func :+1: